rust/lance-core/src/io/object_store.rs

```diff
@@ -224,12 +226,9 @@ impl CredentialProvider for AwsCredentialAdapter {
 /// `credentials_refresh_offset` is the amount of time before expiry to refresh credentials.
 async fn build_aws_credential(
     credentials_refresh_offset: Duration,
-    credentials: Option<Arc<dyn CredentialProvider<Credential = ObjectStoreAwsCredential>>>,
+    credentials: Option<AwsCredentialProvider>,
```

Comment:
We use the dynamic dispatch here: https://github.com/lancedb/lancedb/blob/5ca98c326fdd68a332fd454aaab6d5b6b2f5fc9d/rust/ffi/node/src/lib.rs#L126

But I think this can be replaced with just passing down the appropriate storage options. I think we just use this to pass down explicit user-provided credentials for the node SDK.

cc @chebbyChefNEQ

Reply:
yeah this seems fine. The two types are the same thing https://github.com/apache/arrow-rs/blob/master/object_store/src/aws/mod.rs#L85

Reply:
`AwsCredentialProvider` is just a type alias provided by object_store crate for the same thing

https://github.com/apache/arrow-rs/blob/11b2f5fecc257d97005f2393ee17777ed5d38e7c/object_store/src/aws/mod.rs#L85

Reply:
Oh I see. That makes sense now.